app/controllers/profiles_controller.rb

```diff
@@ -4,15 +4,27 @@
 class ProfilesController < ApplicationController
   def index
     render json: ProfileSerializer.new(
-      policy_scope(Profile.includes(:rules, :hosts))
+      policy_scope(index_relation)
       .paginate(page: params[:page], per_page: params[:per_page])
       .sort_by(&:score)
     )
   end
 
+  def index_relation
+    relation = Profile.includes(:rules, :hosts)
+    unless profile_index_params[:hostname].nil?
```

Comment:
nitpick

```ruby
return (relation = Profile.includes(:rules, :hosts) if profile_index_params[:hostname].blank?
relation.where(hosts: {name: profile_index_params[:hostname]})
```

Reply:
I have two issues with this:

1. It's not extensible. In the future if we decide we want the ability to filter on other columns, we'd have to modify this code rather than adding code. I certainly wanted to write it like this, but see the next point:

```rb
relation.where(hosts: {name: profile_index_params[:hostname]}) unless profile_index_params[:hostname].nil?
```

2. It's not as readable (and it goes over the 80 character line limit - I think rubocop is enforcing that default)